### DISCUSSION TITLE
container stops responding and data gets erased on container restart

### DISCUSSION THREAD

**Discussion by iitbmanish** [polonel/trudesk#647] [General]
Hi. 

I want to give a demo to a client. I have launched an ec2 spot instance in aws. yesterday night i launched and connected to my dns the-funnel.in:8118.  (ip: http://3.111.29.59:8118/)

when i checked in the morning the site was not opening even using the direct ip. So I restarted the pod using 
docker container restart <container id> 

After that it started working but the ids etc that i had created were gone. 

2 problems: why is there no response. 

I saw a similar dicussion but the solution given was check the steps of the docker installation. I did the following during install

git clone https://github.com/polonel/trudesk.git
sudo docker-compose up -d trudesk
sudo docker-compose up -d mongo
sudo docker-compose up -d elasticsearch

let me know if am missing something. Thanks.

**Comment by polonel** [comment]
docker-compose setup is not for any use in production; the documentation states that. It is there to quickly deploy a test site since I removed the online demo site years ago. 

https://docs.trudesk.io/v1.2/getting-started/deployment/docker-deployment

You'll need to deploy using the Manual instructions to store the volumes within the FS and not within docker.

**Comment by iitbmanish** [comment]
followed the steps. however still after 1 day it stopped responding. Also I am getting this error when i am trailing the logs 

01-08-2024 13:49:23 [36] warn:  Cannot read properties of null (reading 'order') - TypeError: Cannot read properties of null (reading 'order')
    at /usr/src/trudesk/src/settings/settingsUtil.js:153:31
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
TypeError: Cannot read properties of null (reading 'order')
    at /usr/src/trudesk/src/settings/settingsUtil.js:153:31
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  level: 'warn',
  [Symbol(level)]: 'warn'
}

**Comment by iitbmanish** [comment]
I was t2.small which had 2 GB ram and 5gb storage. I think that was the problem. Increased it to t3.medium (4 gb ram) and 24 gb storage. Now its working fine.
